tests/test_linear4bit.py

```diff
@@ -0,0 +1,116 @@
+import os
+from contextlib import nullcontext
+from itertools import product
+from tempfile import TemporaryDirectory
+
+import pytest
+import torch
+
+import bitsandbytes as bnb
+from bitsandbytes import functional as F
+from bitsandbytes.nn.modules import Linear4bit
+
+
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="this test requires a GPU")
+@pytest.mark.parametrize(
+    "quant_type, compress_statistics, bias",
+    list(product(["nf4", "fp4"], [False, True], [False, True])),
+)
+def test_linear4_state_dict(quant_type, compress_statistics, bias):
```

Comment:
I think this test looks good. I think another test might be (1) init, (2) copy, (3) quantize, (4) save, (4) del model, (5) load model, (6) compare against reference. This makes sure that we go through the full loop the user is going through. Internally, this should be the same as load/save the state_dict, but it's good to check everything is loaded correctly.

Reply:
I added such test in [my PR to transformers](https://github.com/huggingface/transformers/pull/26037) and it passes together with this PR. I hesitate to bring it to this repo because it would require bringing most of that PR code to the test here. 
Once the transformers PR goes thru -- this would be easy to implement, and I will be happy to make a follow-up PR here.